test/agentchat/test_cache.py

```diff
@@ -0,0 +1,109 @@
+import os
+import sys
+import time
+
+import pytest
+import autogen
+from autogen.agentchat import AssistantAgent, UserProxyAgent
+sys.path.append(os.path.join(os.path.dirname(__file__), ".."))
+from conftest import skip_openai, skip_redis  # noqa: E402
+
+try:
+    from openai import OpenAI
+except ImportError:
+    skip = True
+else:
+    skip = False or skip_openai
+
+try:
+    import redis
+except ImportError:
+    skip_redis = True
+else:
+    skip_redis = False or skip_redis
+
```

Comment:
Can we also have simple mock tests for the cache's public method that does not require openai? OpenAI is expensive and is not required to verify the functionality of the cache

Reply:
implemented test_disk_cache.py and test_redis_cache.py